avaframe/com1DFA/com1DFATools.py

```diff
@@ -44,162 +43,222 @@ def getPartInitMethod(cfg, csz, relThForPart):
-                    "root['GENERAL']['secRelArea']",
-                    "root['GENERAL']['simTypeList']",
-                    "root['INPUT']['releaseScenario']",
-                   ]
+    excludeItems = [
+        "root['GENERAL']['avalancheDir']",
+        "root['GENERAL']['secRelArea']",
+        "root['GENERAL']['simTypeList']",
+        "root['INPUT']['releaseScenario']",
+    ]
 
     # sphKernelSize is set during runtime, make sure it is not reported
     # as changed if default is set to meshCellSize
-    if defCfg['GENERAL']['sphKernelRadius'] == 'meshCellSize':
-        if simCfg['GENERAL']['sphKernelRadius'] == simCfg['GENERAL']['meshCellSize']:
+    if defCfg["GENERAL"]["sphKernelRadius"] == "meshCellSize":
+        if simCfg["GENERAL"]["sphKernelRadius"] == simCfg["GENERAL"]["meshCellSize"]:
             excludeItems.append("root['GENERAL']['sphKernelRadius']")
 
     # do the diff and analyse
     diff = DeepDiff(defCfg, simCfg, exclude_paths=excludeItems)
 
-    if 'values_changed' in diff:
-        log.info('Comparing to default cfg, values changed:')
-        log.info(diff['values_changed'])
-        log.info('Modified from default')
-        isDefault = False
+    # Sometimes (after variation split) the type changes, so check if it is the default or something else
+    # If it is, check the type_changes and convert to values_change if necessary
+    if "type_changes" in diff:
+        diff = _treatTypeChangeDiff(diff)
+
+    valuesChanged = dict()
+    # This needs to be checked AFTER check for type_changes
+    if "values_changed" in diff:
         defaultIdentifierString = 'C'
+        for key, val in diff["values_changed"].items():
+            valuesChanged[_cleanDiffKey(key)] = val
+
+        log.info('Comparing to default cfg, values changed:')
+        log.info(valuesChanged)
+
     else:
-        diff['values_changed'] = None
+        valuesChanged = None
+
+    if "dictionary_item_added" in diff:
+        log.debug("Comparing to default cfg, added items:")
+        log.debug(diff["dictionary_item_added"])
+
+    return defaultIdentifierString, valuesChanged
+
+
+def _treatTypeChangeDiff(diff):
+    """Internal function to convert type changes in the result of DeepDiff to actual changes
+
+    Parameters
+    ----------
+    diff:
+        DeepDiff result to analyse
+
+    Returns
+    -------
+    diff:
+        DeepDiff result with changed values
+    """
+
+    allChangedValues = dict()
 
-    if 'dictionary_item_added' in diff:
-        log.debug('Comparing to default cfg, added items:')
-        log.debug(diff['dictionary_item_added'])
+    for key, val in diff["type_changes"].items():
+        # The type(x)(y) applies the type of x on y and gives the value of y
+        if val["old_value"] == type(val["old_value"])(val["new_value"]):
+            continue
+        else:
+            del val["old_type"]
+            del val["new_type"]
+            val["new_value"] = type(val["old_value"])(val["new_value"])
```

Comment:
if the val['old_type'] and next line are deleted, do they still exist in line 189?

Reply:
They don't need to, since in line 189 old_value is queried, not old_type